File: codes/js/problems/128/101.js
/**
 * @param {TreeNode} l
 * @param {TreeNode} r
 * @return {boolean}
 */
const check = (l, r) => {
  if (!l && !r) return true
  if (!l || !r) return false
  return l.val === r.val && check(l.left, r.right) && check(r.left, l.right)
}

/**
 * @param {TreeNode} root
 * @return {boolean}
 */
var isSymmetric = function(root) {
  return check(root.left, root.right)
}

File: py3/128/101.py
# Definition for a binary tree node.
class TreeNode:
    def __init__(self, x):
        self.val = x
        self.left = None
        self.right = None

class Solution:
    def isSymmetric(self, root: TreeNode) -> bool:
        def isEqual(A: TreeNode, B: TreeNode) -> bool:
            if not A and not B: return True
            if not A or not B or A.val != B.val: return False
            return isEqual(A.left, B.right) and isEqual(A.right, B.left)

        return isEqual(root.left, root.right) if root else True